### PR TITLE
Arcade - Add new keywords and function names to be on par with version 1.24

### LINKED ISSUE #3873: (Arcade) Add new keywords that will be included in next version (1.24)
**Describe the issue**

[Arcade](https://developers.arcgis.com/arcade/) has added new keywords and built in functions over the years. The current list is out of date. The following now represents the complete list of keywords:

**New Keywords**

- continue
- in
- case
- catch
- debugger
- export
- import
- switch
- try

**literals**

n/a

**built_in**

This is the complete list of built in functions. 

```
[
  "Abs",
  "Acos",
  "All",
  "Angle",
  "Any",
  "Area",
  "AreaGeodetic",
  "Array",
  "Asin",
  "Atan",
  "Atan2",
  "Attachments",
  "Average",
  "Back",
  "Bearing",
  "Boolean",
  "Buffer",
  "BufferGeodetic",
  "Ceil",
  "Centroid",
  "ChangeTimeZone",
  "Clip",
  "Concatenate",
  "Console",
  "Constrain",
  "Contains",
  "ConvertDirection",
  "ConvexHull",
  "Cos",
  "Count",
  "Crosses",
  "Cut",
  "Date",
  "DateAdd",
  "DateDiff",
  "DateOnly",
  "Day",
  "Decode",
  "DefaultValue",
  "Densify",
  "DensifyGeodetic",
  "Dictionary",
  "Difference",
  "Disjoint",
  "Distance",
  "DistanceGeodetic",
  "Distinct",
  "Domain",
  "DomainCode",
  "DomainName",
  "EnvelopeIntersects",
  "Equals",
  "Erase",
  "Exp",
  "Expects",
  "Extent",
  "Feature",
  "FeatureSet",
  "FeatureSetByAssociation",
  "FeatureSetById",
  "FeatureSetByName",
  "FeatureSetByPortalItem",
  "FeatureSetByRelationshipClass",
  "FeatureSetByRelationshipName",
  "Filter",
  "Find",
  "First",
  "Floor",
  "FromCharCode",
  "FromCodePoint",
  "FromJSON",
  "Front",
  "GdbVersion",
  "Generalize",
  "Geometry",
  "GetEnvironment",
  "GetFeatureSet",
  "GetFeatureSetInfo",
  "GetUser",
  "GroupBy",
  "Guid",
  "HasKey",
  "HasValue",
  "Hash",
  "Hour",
  "IIf",
  "ISOMonth",
  "ISOWeek",
  "ISOWeekday",
  "ISOYear",
  "Includes",
  "IndexOf",
  "Insert",
  "Intersection",
  "Intersects",
  "IsEmpty",
  "IsNan",
  "IsSelfIntersecting",
  "IsSimple",
  "Left",
  "Length",
  "Length3D",
  "LengthGeodetic",
  "Log",
  "Lower",
  "Map",
  "Max",
  "Mean",
  "Mid",
  "Millisecond",
  "Min",
  "Minute",
  "Month",
  "MultiPartToSinglePart",
  "Multipoint",
  "NearestCoordinate",
  "NearestVertex",
  "NextSequenceValue",
  "None",
  "Now",
  "Number",
  "Offset",
  "OrderBy",
  "Overlaps",
  "Point",
  "Polygon",
  "Polyline",
  "Pop",
  "Portal",
  "Pow",
  "Proper",
  "Push",
  "Random",
  "Reduce",
  "Relate",
  "Replace",
  "Resize",
  "Reverse",
  "Right",
  "RingIsClockwise",
  "Rotate",
  "Round",
  "Schema",
  "Second",
  "SetGeometry",
  "Simplify",
  "Sin",
  "Slice",
  "Sort",
  "Splice",
  "Split",
  "Sqrt",
  "StandardizeGuid",
  "Stdev",
  "SubtypeCode",
  "SubtypeName",
  "Subtypes",
  "Sum",
  "SymmetricDifference",
  "Tan",
  "Text",
  "Time",
  "TimeZone",
  "TimeZoneOffset",
  "Timestamp",
  "ToCharCode",
  "ToCodePoint",
  "ToHex",
  "ToLocal",
  "ToUTC",
  "Today",
  "Top",
  "Touches",
  "TrackAccelerationAt",
  "TrackAccelerationWindow",
  "TrackCurrentAcceleration",
  "TrackCurrentDistance",
  "TrackCurrentSpeed",
  "TrackCurrentTime",
  "TrackDistanceAt",
  "TrackDistanceWindow",
  "TrackDuration",
  "TrackFieldWindow",
  "TrackGeometryWindow",
  "TrackIndex",
  "TrackSpeedAt",
  "TrackSpeedWindow",
  "TrackStartTime",
  "TrackWindow",
  "Trim",
  "TypeOf",
  "Union",
  "Upper",
  "UrlEncode",
  "Variance",
  "Week",
  "Weekday",
  "When",
  "Within",
  "Year",
];
```

**Profile variables**

[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view]

The following are profile variables preceded by the `$` character that are excluded from the list above:

- $editcontext
- $originalFeature
- $featureSet
- $datapoint
- $reference
- $rowindex
- $record
- $config
- $sourcedatastore
- $targetdatastore
- $feedfeature
- $fencefeature
- $fencenotificationtype
- $locationupdate
- $measure
- $aggregatedFeatures
- $target
- $join
- $analytic

I can submit a PR for this.

### PR DESCRIPTION
**Pull request by ekenes**
Resolves #3873

### Changes

Adds all the keywords, functions, and profile names listed in #3873 for proper syntax highlighting.

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

File: src/languages/arcade.js [hunk 1]
```diff
@@ -223,14 +246,14 @@ export default function(hljs) {
...
+      "When|0",
       "Within",
-      "Year"
+      "Year|0",
     ]
   };
   const SYMBOL = {
     className: 'symbol',
-    begin: '\\$[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view]+'
+    begin: '\\$[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view|editcontext|originalFeature|featureSet|datapoint|reference|rowindex|record|config|sourcedatastore|targetdatastore|feedfeature|fencefeature|fencenotificationtype|locationupdate|measure|aggregatedFeatures|target|join|analytic]+'
```
Comment (joshgoebel) [inline src/languages/arcade.js:256]: I think we want `()` here, not `[]`, do we not?  (yes, I realize it's a mistake that predates your PR).   So we need to fix at least that.

Optimally we'd first declared the list as an array (one item per line) and then use `regex.either` to pull it together in regex form... I'll let you decide if you want to complete that work - you can find examples in other grammars.

**Comment by joshgoebel** [comment]
Thanks, this will be awesome once we iron out the one regex issue!

**Comment by ekenes** [comment]
Thanks @joshgoebel! I just updated to use `regex.either`. I'm not sure if I used it correctly, but the previous tests I added still pass. Let me know how it looks.

File: src/languages/arcade.js [hunk 2]
```diff
@@ -251,9 +252,41 @@ export default function(hljs) {
...
+    "record",
+    "config",
+    "sourcedatastore",
+    "targetdatastore",
+    "feedfeature",
+    "fencefeature",
+    "fencenotificationtype",
+    "locationupdate",
+    "measure",
+    "aggregatedFeatures",
```
Comment (joshgoebel) [inline src/languages/arcade.js:282]: Any reason not to alpha sort these?

**Comment by joshgoebel** [comment]
> I'm not sure if I used it correctly,

Looked good to me.

**Review by joshgoebel — APPROVED** [review]

**Comment by ekenes** [comment]
@joshgoebel This was listed under the 11.9 changelog. However, I saw that 11.9 published a couple weeks ago. Which version should I look out for this in?

**Comment by joshgoebel** [comment]
Oh sorry I didn't catch this in final review... that changelog entry will need to be moved to 11.10, since 11.9 has passed.

**Pull request merged**